File: Atcoder/abc357/D.py
def qpow(a, b, p):
    """快速幂函数，计算 a^b % p"""
    ans = 1
    while b:
        if b & 1:
            ans = ans * a % p
        b >>= 1
        a = a * a % p
    return ans

def inv(a, p):
    """模逆元，计算 a^(p-2) % p"""
    return qpow(a, p-2, p)

def main():
    N =  int(input())

    p = 998244353

    # 计算 w，w 是 N 的十进制位数
    w = len(str(N))

    # 计算 (10^(w*N) - 1) % p
    term1 = (qpow(10, w * N, p) - 1 + p) % p

    # 计算 (10^w - 1) % p
    term2 = (qpow(10, w, p) - 1 + p) % p

    # 计算 term2 的模逆元
    term2_inv = inv(term2, p)

    # 计算最终结果
    result = (term1 * term2_inv %p * N) % p

    print(result)

if __name__ == "__main__":
    main()

File: Atcoder/abc357/D.cpp
/**************************************************************
 * Problem: 
 * Author: Vanilla_chan
 * Date: 
 * E-Mail: heshaohong2015@outlook.com
 **************************************************************/
#include<iostream>
#include<algorithm>
#include<cstdio>
#include<string>
#include<cstring>
#include<cmath>
#include<map>
#include<set>
#include<queue>
#include<vector>
#include<limits.h>
#define IL inline
#define re register
#define LL long long
#define ULL unsigned long long
#ifdef TH
#define debug printf("Now is %d\n",__LINE__);
#else
#define debug
#endif
#ifdef ONLINE_JUDGE
char buf[1<<23],* p1=buf,* p2=buf,obuf[1<<23],* O=obuf;
#define getchar() (p1==p2&&(p2=(p1=buf)+fread(buf,1,1<<21,stdin),p1==p2)?EOF:*p1++)
#endif
using namespace std;




#define p 998244353ull

ULL qpow(ULL a,ULL b){
	a%=p;
	ULL ans=1;
	while(b){
		if(b&1) ans=ans*a%p;
		b>>=1;
		a=a*a%p;
	}
	return ans;
}

ULL inv(ULL a){
	return qpow(a%p,p-2)%p;
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);
	cout.precision(10);
	int t=1;
//	cin>>t;
	while(t--)
	{
		ULL N,w;
		cin>>N;
		w=to_string(N).size();
		//w=log10(N)+1;
		cout<< N %p *(qpow(qpow(10,w),N)-1+p) %p * inv((qpow(10,w)-1+p)%p) %p;
	}
	return 0;
}
